test/udfs/test_abstract_udf.py

```diff
@@ -0,0 +1,107 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import inspect
+import unittest
+from enum import Enum
+from inspect import isabstract
+from test.util import get_all_subclasses, get_mock_object
+
+import eva
+from eva.udfs.abstract.abstract_udf import AbstractUDF
+
+
+class AbstractUDFTest(unittest.TestCase):
+    def test_udf_abstract_functions(self):
+        derived_udf_classes = list(get_all_subclasses(AbstractUDF))
```

Comment:
This is quite complex logic. Can you drop some comments or add references? Thanks!

Reply:
Yes, added comments and references.